test/multiverse/suites/elasticsearch/elasticsearch_instrumentation_test.rb

```diff
@@ -42,9 +40,26 @@ def test_segment_elasticsearch_product
     assert_equal NewRelic::Agent::Instrumentation::Elasticsearch::PRODUCT_NAME, @segment.product
   end
 
-  def test_segment_operation
+  def test_segment_operation_matches_method_called
     search
-    assert_equal NewRelic::Agent::Instrumentation::Elasticsearch::OPERATION, @segment.operation
+    assert_equal 'search', @segment.operation
+  end
+
+  def test_segment_operation_matches_method_called_when_indexing
+    txn = in_transaction do
+      @client.index(index: 'my-index', id: 1, body: {title: 'Test'})
+    end
+
+    segment = txn.segments[1]
+    assert_equal 'index', segment.operation
+  end
+
+  def test_segment_operation_returns_OPERATION_when_method_nil
+    skip('How can I stub only the Array#index method?')
```

Comment:
I'm assuming this means that the following line trying to stub isn't working how you want. I'm not sure how to do it with minitest stubs as I'm less familiar with that, but maybe James has some ideas since he's more familiar that.
I've used mocha more often so I'm more familiar with it. I'm pretty sure we can do this with mocha stubs though, using any_instance. 
`Array.any_instance.stubs(:index).returns(nil)`
any_instance isn't really recommended generally, but we still do use it quite a lot in the agent tests already. If we do it this way, we'd also want to make sure `mocha_teardown` gets added to the teardown method in here as well, so the stub doesn't leak past the test.

Reply:
Oh, I'm bummed I missed this until now! I readjusted the code and the tests a bit to make this case work in a different way. I'd be curious to see what you think!

Reply:
oh its cool, I'm glad you figured out another way to test it!